### PR TITLE
feat: impl `DealProposal.cid()`

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- Impl `DealProposal.cid()` which is need for nv17 state migration, see [code](https://github.com/filecoin-project/go-state-types/blob/master/builtin/v9/migration/util.go#L98)
- Add go parity unit test for `fn cid`

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: actors/market/src/v8/deal.rs [hunk 1]
```diff
@@ -123,6 +124,13 @@ impl DealProposal {
     pub fn provider_balance_requirement(&self) -> &TokenAmount {
         &self.provider_collateral
     }
+    pub fn cid(&self) -> anyhow::Result<Cid> {
```
Comment (lemmih) [inline actors/market/src/v8/deal.rs:127]: Maybe we want to implement `fvm_ipld_encoding::Cbor` for `DealProposal`. The `Cbor` trait also defines `cid()`.

Comment (hanabi1224) [reply actors/market/src/v8/deal.rs:127]: It seems to be deprecated now: https://docs.rs/fvm_ipld_encoding/latest/fvm_ipld_encoding/trait.Cbor.html

**Review by lemmih — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,33 @@
+module test/v2
```
Comment (LesnyRumcajs) [inline go.mod:1]: Could we have those Go files in some tests/dev directory? They may be confusing for anyone visiting the repository.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ on:
     branches:
       - main
     tags:
-      - 'v*'
+      - "v*"
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply .github/workflows/tests.yml:17]: It's from my IDE auto-formatter

File: actors/market/src/v8/deal.rs [hunk 4]
```diff
@@ -141,3 +149,95 @@ pub struct DealState {
...
+    use anyhow::*;
+    use pretty_assertions::assert_eq;
+    use quickcheck_macros::quickcheck;
+
+    use super::*;
+
+    const GO_TEST_DIR: &str = "tests/go";
+
+    #[quickcheck]
+    fn test_deal_proposal_cid(proposal: DealProposal) -> Result<()> {
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply actors/market/src/v8/deal.rs:190]: Just checked the source code, test count can be controlled by env var `QUICKCHECK_TESTS` and `QUICKCHECK_MAX_TESTS`, but setting these env vars would effect all quickcheck tests, sounds good?

https://github.com/BurntSushi/quickcheck/blob/master/src/tester.rs#L21

File: go.mod (hunk 2 above)
Comment (hanabi1224) [reply go.mod:1]: It has to be parent dir of the go source file, what folder structure would u suggest?

File: actors/market/src/v8/deal.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (hanabi1224) [reply actors/market/src/v8/deal.rs:190]: Now it's ~5s 
https://github.com/ChainSafe/fil-actor-states/actions/runs/5067045084/jobs/9097597596?pr=120#step:6:310

File: go.mod (hunk 2 above)
Comment (LesnyRumcajs) [reply go.mod:1]: How about we put all the go tests in `go_compat/tests/` and those mod files in `go_compat/`? Would that work?

Comment (hanabi1224) [reply go.mod:1]: That works but making it a bit hard to locate the file from rust code. Maybe worth adding some test utilities

Comment (hanabi1224) [reply go.mod]: Fixed

**Review by LesnyRumcajs — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (LesnyRumcajs) [reply go.mod]: rock-solid, thanks!

**Pull request merged**